### PR TITLE
Fixing NaN caused by batch-norm for theano Update theano_backend.py

### PR DESCRIPTION
**Pull request by EdwardRaff**
fix batch norm causing NaN for many datasets, fixes issue #3475

### CODE REVIEW

**Comment by fchollet** [comment]
LGTM, thanks.
